### ISSUE TITLE
[BUG] using an empty compose file causes project name to be unset

### ISSUE THREAD

**Issue by designermonkey** [docker/compose#11450]
### Description

I have a command which concatenates multiple docker-compose yaml files, like so:
```
docker compose -f ./docker-compose.yml -f ./docker-compose.dev.yml -f ./docker-compose.uat.yml build
```

Because of the way our local build system works, the last file is empty until we define specifics for our UAT process. We have defined the `name` top-level element in the initial `docker-compose.yml` file.

Using this empty placeholder file at the end of the list seems to unset content from the other files resulting in the following:
```
project name can't be empty. Use `--project-name` to set a valid name
```

I can't check whether it is unsetting other content as well

### Steps To Reproduce

Use multiple compose files with content, except the last file which is empty:
```
docker compose -f ./docker-compose.yml -f ./docker-compose.dev.yml -f ./docker-compose.uat.yml build
```

### Compose Version

v2.24.3-desktop.1

### Docker Environment

Docker desktop: 4.27.1 (136059) on Mac OS 14.2.1 (23C71)

### Anything else?

I accidentally pressed enter while filling out the form, if anything is missing, please let me know.

**Comment by designermonkey** [comment]
If I add `---` to the top of the empty file, I get `Top-level object must be a mapping` instead.

Shouldn't an empty yaml file still be considered valid yaml content? (I don't actually know).

**Comment by milas** [comment]
This fix has been included in Compose [2.24.7+](https://github.com/docker/compose/releases/latest). If you continue to have problems after upgrading, comment here and we can re-open the issue or create a new one as appropriate. Thanks for the report!

**Comment by digglife** [comment]
Thanks. Can't update for some reason so I added `name: foo` to the empty yaml.
